distributed/worker.py

```diff
@@ -3733,6 +3733,7 @@ def _ensure_computing(self) -> RecsInstrs:
     async def execute(self, key: str, *, stimulus_id: str) -> StateMachineEvent | None:
         if self.status in {Status.closing, Status.closed, Status.closing_gracefully}:
             return None
+        await self._started.wait()
```

Comment:
This seems like an odd/late place to be waiting. I would think that it should be the responsibility of things far upstream of `execute` to wait to prevent us from getting here. Like maybe `Server.handle_stream` should wait to process messages until `_started` has been set, or something like that.

If this is here, why shouldn't it be in basically every async method? Or at least also in `gather_dep`? This feels like something that ideally `Server` subclasses shouldn't have to worry about.

Reply:
OK, so given that I'm unlikely to explore this much more deeply, what do you think we should do here process-wise?  

Should we close this PR and put this on the backlog to be prioritized more properly?

Reply:
I think close this, open an issue, and then put that on the backlog. There are a few levels of issues:
a. Worker can start executing tasks before plugins are started
b. Worker can generally start doing things (executing tasks, fetching deps, etc.) before it's fully started
c. Scheduler generally may think a worker has started while it's still starting up, which could lead to poor scheduling decisions (if a plugin is going to take 5min to load some data before the worker is "started", don't schedule tasks on it yet). I don't think we use `Status.starting` enough. Probably a lot of the logic currently in `Scheduler.add_worker` should actually be called from `handle_worker_status_change`, when the worker switches to `running` from `starting`.

Some of these we'll address and prioritize above others.

Reply:
I agree that we should raise an issue.  I disagree that we should close this.  I think that this is a strict improvement over current state, and handles 90% of the likely issues (admittedly subjective).